src/Console/BaseCommand.php

```diff
@@ -0,0 +1,271 @@
+     */
+    protected function buildOptionParser(ConsoleOptionParser $parser): ConsoleOptionParser
+    {
+        return $parser;
+    }
+
+    /**
+     * Hook method invoked by CakePHP when a command is about to be executed.
+     *
+     * Override this method and implement expensive/important setup steps that
+     * should not run on every command run. This method will be called *before*
+     * the options and arguments are validated and processed.
+     *
+     * @return void
+     */
+    public function initialize(): void
+    {
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function run(array $argv, ConsoleIo $io): ?int
+    {
+        $this->initialize();
+
+        $parser = $this->getOptionParser();
+        try {
+            [$options, $arguments] = $parser->parse($argv);
+            $args = new Arguments(
+                $arguments,
+                $options,
+                $parser->argumentNames()
+            );
+        } catch (ConsoleException $e) {
+            $io->err('Error: ' . $e->getMessage());
+
+            return static::CODE_ERROR;
+        }
+        $this->setOutputLevel($args, $io);
+
+        if ($args->getOption('help')) {
+            $this->displayHelp($parser, $args, $io);
+
+            return static::CODE_SUCCESS;
+        }
+
+        return $this->execute($args, $io);
+    }
+
+    /**
+     * Output help content
+     *
+     * @param \Cake\Console\ConsoleOptionParser $parser The option parser.
+     * @param \Cake\Console\Arguments $args The command arguments.
+     * @param \Cake\Console\ConsoleIo $io The console io
+     * @return void
+     */
+    protected function displayHelp(ConsoleOptionParser $parser, Arguments $args, ConsoleIo $io): void
+    {
+        $format = 'text';
+        if ($args->getArgumentAt(0) === 'xml') {
+            $format = 'xml';
+            $io->setOutputAs(ConsoleOutput::RAW);
+        }
+
+        $io->out($parser->help(null, $format));
+    }
+
+    /**
+     * Set the output level based on the Arguments.
+     *
+     * @param \Cake\Console\Arguments $args The command arguments.
+     * @param \Cake\Console\ConsoleIo $io The console io
+     * @return void
+     */
+    protected function setOutputLevel(Arguments $args, ConsoleIo $io): void
+    {
+        $io->setLoggers(ConsoleIo::NORMAL);
+        if ($args->getOption('quiet')) {
+            $io->level(ConsoleIo::QUIET);
+            $io->setLoggers(ConsoleIo::QUIET);
+        }
+        if ($args->getOption('verbose')) {
+            $io->level(ConsoleIo::VERBOSE);
+            $io->setLoggers(ConsoleIo::VERBOSE);
+        }
+    }
+
+    /**
+     * Implement this method with your command's logic.
+     *
+     * @param \Cake\Console\Arguments $args The command arguments.
+     * @param \Cake\Console\ConsoleIo $io The console io
+     * @return int|null The exit code or null for success
+     */
+    public function execute(Arguments $args, ConsoleIo $io): ?int
```

Comment:
since we have `command` class that extended this class 
probably you can define  this method as abstract

Reply:
Good idea :+1: